tests/lexer/mlexerutils.nim

```diff
@@ -0,0 +1,14 @@
+import macros
+
+macro lispReprStr*(a: untyped): untyped = newLit(a.lispRepr)
+
+macro assertAST*(expected: string, struct: untyped): untyped =
+  var ast = newLit(struct.treeRepr)
+  result = quote do:
+    if `ast` != `expected`:
+      echo "Got:"
+      echo `ast`.indent(2)
+      echo "Expected:"
+      echo `expected`.indent(2)
+      raise newException(ValueError, "Failed to lex properly")
```

Comment:
```nim
macro assertAST*(expected: string, struct: untyped): untyped =
  var ast = newLit(struct.treeRepr)
  result = quote do:
    if `ast` != `expected`:
      doAssert false, "\nGot:\n" & `ast`.indent(2) & "\nExpected:\n" & `expected`.indent(2)
```

* echo + raise is bad as catcher won't get the full msg.
* splitting echo is bad because it increases chance a msg gets split and intermingled with other processes' stdout when multiple processes are outputting to stdout as is case with `testament pcat`